### DISCUSSION TITLE
NoSQL 도입에 대해서

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#344] [개발 논의]
# 서론

사용자의 사전 식물 등록 요청 구현과정에서 다음과 같은 데이터가 발생하게 됩니다.

- 요청 식물명
- 참고 이미지

해당 값들은 nullable한 값이며 RDB 상에서 다른 테이블과 연관관계를 가지지 않습니다.

# 문제인식

해당 데이터는 정규화된 데이터가 아닌 값이며 사용자에게 보여지는 데이터도 아닙니다.
이로부터 다음과 같은 문제점이 생각납니다.

- 관리 목적의 데이터가 RDB에 쓰기연산을 수행하면서 운영 DB의 성능에 영향을 미친다. (프로덕션 DB 운영 간 불필요한 연산이 수행됨)

# 제안

이러한 문제점을 해결하기 위해 NoSQL을 도입하는 것에 대해 제안드립니다.

key-value 형태로 데이터를 저장하여 데이터의 정규화에 리소스를 소모하지 않아도 된다는 부분이 가장 큰 장점이라고 생각되는데 여러분들의 의견이 궁금합니다.

@rawfishthelgh @Kim0914 @yeonkkk

**Comment by yeonkkk** [comment]
DB 분리 및 NoSQL에 찬성합니다! 

주어진 시간이 많지 않은 현재 상황에서 `key-value 형태로 데이터를 저장하여 데이터의 정규화에 리소스를 소모하지 않아도 된다`는 부분이 큰 장점으로 와닿네요.
뿐만 아니라 현재는 없는 기능이지만 이후 등록 요청 데이터를 이용해서 바로 사전 식물을 등록하는 기능에 대해 고려한다면, 필드 추가와 데이터 변경이 많이 필요할 것 같습니다. 
당장 적용할 기능은 아니지만, 어차피 DB를 분리할거라면 확장성을 생각해서 NoSQL을 도입하는 것이 좋아보입니다.

**Comment by Choi-JJunho** [comment]
RDB 상에 저장하는 방식으로 구현하여 해당 discussion close 하겠습니다.

관련 PR
- https://github.com/woowacourse-teams/2023-pium/pull/367
